lib/ansible/modules/utilities/logic/wait_for_connection.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2017, Dag Wieers <dag@wieers.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'status': ['stableinterface'],
+                    'supported_by': 'core',
+                    'version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: wait_for_connection
+short_description: Waits until remote system is reachable/usable
+description:
+- Waits for a total of C(timeout) seconds.
+- Retries the transport connection after a timeout of C(connect_timeout).
+- Tests the transport connection every C(sleep) seconds.
+- This module makes use of internal ansible transport (and configuration) and the ping/win_ping module to guarantee correct end-to-end functioning.
+version_added: "2.3"
+options:
+  connect_timeout:
+    description:
+      - Maximum number of seconds to wait for a connection to happen before closing and retrying.
+    default: 5
+  delay:
+    description:
+      - Number of seconds to wait before starting to poll.
+    default: 0
+  sleep:
+    default: 1
+    description:
+      - Number of seconds to sleep between checks.
+  timeout:
+    description:
+      - Maximum number of seconds to wait for.
+    default: 300
+author: "Dag Wieers (@dagwieers)"
+'''
+
+EXAMPLES = '''
+- name: Wait 300 seconds for target connection to become reachable/usable
+  wait_for_connection:
+
+- name: Wait 600 seconds, but only start checking after 60 seconds
+  wait_for_connection:
+    delay: 60
+    timeout: 600
+
+# Build a new VM, wait for it to become ready and continue playbook
+- hosts: all
+  gather_facts: no
+  tasks:
+  - name: Clone new VM, if missing
+    vmware_guest:
+      hostname: '{{ vcenter_ipaddress }}'
+      ...
+      name: '{{ inventory_hostname_short }}'
+      template: Windows 2012R2
+      state: present
+      networks:
+      - name: VLAN 123
+        ip: '{{ ansible_host }}'
+        netmask: '{{ vm_netmask }}'
+        gateway: '{{ gw_ip }}'
+        domain: '{{ domain }}'
+        dns_servers: [ '{{ dns_ip }}' ]
+      customization:
+        hostname: '{{ vm_shortname }}'
+        runonce:
+        - powershell.exe -ExecutionPolicy Unrestricted -File C:\Windows\Temp\Enable-WinRM.ps1 -CertValidityDays 3650 -ForceNewSSLCert
```

Comment:
I know you aren't fond of the current name, but for clarity I'd consider referring to the ConfigureRemotingForAnsible.ps1 name here